src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Consider adding a default case in the switch statement for error handling, as per the checklist item: 'switch/case should always have default case for error handling.'